BUG: restrict the __config__ modifications to win32

xoviat opened the pull request:
It seems that sadly 3051ccff47cf3f55ec7434d5b787bd0c1e08c012 had the unintended side-effect of exposing buggy code in that such code should only be run on win32.

eric-wieser commented on numpy/distutils/misc_util.py:
```diff
@@ -2285,8 +2285,10 @@ def generate_config_py(target):
 extra_dll_dir = os.path.join(os.path.dirname(__file__), '.libs')
-if os.path.isdir(extra_dll_dir):
+if sys.platform == 'win32' and os.path.isdir(extra_dll_dir):
     os.environ["PATH"] += os.pathsep + extra_dll_dir
```
Wouldn't it be better do the right thing even when `PATH` is not set? So:

```
os.environ["PATH"] = os.pathsep.join(os.environ.get("PATH").split(os.pathsep) + [extra_dll_dir])
```

xoviat commented:
Note: this needs a forwardport.

charris commented on numpy/distutils/misc_util.py:
```diff
@@ -2285,9 +2285,11 @@ def generate_config_py(target):
-if os.path.isdir(extra_dll_dir):
-    os.environ["PATH"] += os.pathsep + extra_dll_dir
+if sys.platform == 'win32' and os.path.isdir(extra_dll_dir):
+    os.environ["PATH"] = os.pathsep.join(os.environ.get("PATH", "").split(os.pathsep) + [extra_dll_dir])
```
100 char > 79 char. Why do you split the path?

charris replied:
Note that with the refault path `""`, you still end up with a `os.pathsep` prefixing the `extra_dll_dir`. Is that OK?

xoviat replied:
Suggested by @eric-wieser. I think I like the previous version better.

eric-wieser replied:
> Why do you split the path?

An attempt to not end up with that extra pathsep. I thought that `"".split(":")` would return the empty list, but it does not.

xoviat committed:
BUG: restrict the __config__ modifications to win32

It seems that sadly 3051ccff47cf3f55ec7434d5b787bd0c1e08c012 had the unintended
side-effect of exposing buggy code that should only be
run on win32.

xoviat commented:
I think that's okay?

eric-wieser commented:
I don't think you meant to link to 3051ccf, which doesn't touch the conditional

xoviat commented:
Yes, I did. I thought it was only a name change, so the change would not matter, but it exposed the assumption that PATH is always present to all platforms rather than just win32, leading to the bug report. The change itself wasn't a bad change, but ideally I would have fixed this line as well in that PR.

xoviat commented:
FWIW, it just never occurred to me that PATH would not be set.

charris commented:
Thanks @xoviat.

xoviat commented:
Hopefully this is the last that we will need to modify this.
